pkg/alertmanager/operator.go

```diff
@@ -654,6 +658,19 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	level.Info(logger).Log("msg", "sync alertmanager")
 
+	// Check if operator has the permission to read the storage class.
+	if c.canReadStorageClass {
+		// Check if the AlertManager Specs is using a Volume Claim.
+		if !reflect.DeepEqual(am.Spec.Storage.VolumeClaimTemplate, monitoringv1.EmbeddedPersistentVolumeClaim{}) {
+			storageClassName := am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName
+			_, err := c.kclient.StorageV1().StorageClasses().Get(ctx, *storageClassName, metav1.GetOptions{})
+			if err != nil {
```

Comment:
If an non-nil error is returned, we need to distinguish between not found and other errors.
```suggestion
			if err != nil {
				if apierrors.IsNotFound{
```

Reply:
Sure, Updating the code.